src/resolve-dependency.ts

```diff
@@ -123,32 +127,65 @@ function getExportsTarget(exports: string | string[] | { [key: string]: string }
   return undefined;
 }
 
-function resolveExportsTarget (pkgPath: string, exp: string | string[] | { [key: string]: string }, subpath: string, job: Job, cjsResolve: boolean): string | undefined {
-  let exports: { [key: string]: string | string[] | { [key: string]: string } };
-  if (typeof exp === 'string' ||
-      typeof exp === 'object' && !Array.isArray(exp) && Object.keys(exp).length && Object.keys(exp)[0][0] !== '.') {
-    exports = { '.' : exp };
+function resolveExportsImports (pkgPath: string, obj: PackageTarget, subpath: string, job: Job, isImports: boolean, cjsResolve: boolean): string | undefined {
+  let matchObj: { [key: string]: PackageTarget };
+  if (isImports) {
+    if (!(typeof obj === 'object' && !Array.isArray(obj) && obj !== null))
+      return undefined;
+    matchObj = obj;
+  } else if (typeof obj === 'string' || Array.isArray(obj) || obj === null ||
+      typeof obj === 'object' && Object.keys(obj).length && Object.keys(obj)[0][0] !== '.') {
+    matchObj = { '.' : obj };
   } else {
-    exports = exp;
+    matchObj = obj;
   }
   
-  if (subpath in exports) {
-    const target = getExportsTarget(exports[subpath], job.exports, cjsResolve);
+  if (subpath in matchObj) {
+    const target = getExportsTarget(matchObj[subpath], job.conditions, cjsResolve);
     if (typeof target === 'string' && target.startsWith('./'))
       return pkgPath + target.slice(1);
   }
-  for (const match of Object.keys(exports)) {
+  for (const match of Object.keys(matchObj).sort((a, b) => a.length > b.length ? -1 : 1)) {
```

Comment:
```suggestion
  for (const match of Object.keys(matchObj).sort((a, b) => b.length - a.length)) {
```

Slightly shorter way to do the same